### ISSUE TITLE
BUG: Theme is broken after active custom theme is deleted

### ISSUE THREAD

**Issue by svreic** [oqtane/oqtane.framework#633]
**Scenario**
If the currently used custom theme is deleted than the theme of the site that used the custom theme is broken. 

**Steps to reproduce**

1. Install a custom external theme (e.g. the cerulean theme)
2. Go to site management and use this custom theme as default theme for the site (also apply layout and container)
3. Reload the page -> new theme is used
4. Go to theme management and delete the custom theme
5. Reload the page -> default oqtane theme is loaded but is broken (see screenshot)

**Expected Behavior**

- The default theme should look like expected.

**Actual Behavior**

- The default theme is broken.

**Screenshot**

![image](https://user-images.githubusercontent.com/64966674/85289067-214d5900-b497-11ea-823e-a9c036c6a498.png)
**Reason**

The reason why the theme is broken is that the theme specific css is not loaded correctly. And this is a result of how the theme is applied to a page.  If a page is requested than the SiteRouter processes this page. Within this processing the theme of the page is determined. Most of the pages don't have an own theme so the site theme is used. Wich theme should be used is read from the db as a string. But the entry in the db is already the name of the "old" custom theme and not the name of the default theme! 

The following code in the ProcessPage method of the SiteRouter is trying to get the type for the theme name that is defined in page.ThemeType.
``` csharp
Type themetype = Type.GetType(page.ThemeType);
if (themetype != null)
{
    var themeobject = Activator.CreateInstance(themetype) as IThemeControl;
    if (themeobject != null)
    {
        panes = themeobject.Panes;
        page.Resources = ManagePageResources(page.Resources, themeobject.Resources);
    }
}
```
The first line returns null because no type for the old custom theme name exists anymore. The code is implemented defensive so there is no error at this point and the page. The page.Resources collection is not filled with the needed information.

Within the ThemeBuilder component this "wrong" resources collection is used - among other things - to add needed stylesheets for the theme.

**Possible solutions**

Imho there are 3 possible solutions:
1. The theme management component could test all deletable themes whether they are used in a site or page and disable the delete button if this is the case. 
2. If the delete button of a theme is clicked the component tests whether this theme is used anywhere. If this is the case the user should be informed and all references to the theme should be changed to the default oqtane theme. This ensures the integrity of the data in the db.
3. At the point in the ThemeBuilder component where the default theme is used as a fallback if the given theme is not found the adding of the css files to the header is processed.

For me option 2 is the favorite. Option 1 could be very time consuming depending on the number of themes, sites and pages. Option 3 however feels more like a workaround because it does not really address the root cause of the problem.

**Comment by svreic** [comment]
@sbwalker please can you give a statement about the suggested solutions? Or maybe there is a way better solution than the ones i had in mind.

**Comment by sbwalker** [comment]
Thank you for the detailed analysis - I should be able to provide my feedback today

**Comment by sbwalker** [comment]
I believe this was fixed in #651
